CactbotOverlay/FFXIVProcess.cs

```diff
@@ -676,25 +632,95 @@ public class PaladinJobData {
       }
     }
 
-    public PaladinJobData GetPaladin() {
-      byte[] bytes = GetJobSpecificData();
-      if (bytes == null)
-        return null;
+    public unsafe PaladinJobData GetPaladin() {
+      fixed (byte* p = GetJobSpecificData()) {
+        if (p == null)
+          return null;
+        else {
+          PaladinJobMemory mem = *(PaladinJobMemory*)&p[0];
+          PaladinJobData j = new PaladinJobData() {
+            oath = mem.oath,
+          };
+          return j;
+        }
+      }
+    }
+
+    [StructLayout(LayoutKind.Explicit)]
+    public struct GunbreakerJobMemory { // TODO: Needs a lot more research. 
```

Comment:
I'm not familiar with gunbreaker's mechanics and I don't have it levelled to be able to work out how this data works. I've only been able to make a guess.

Reply:
I have gnb at 80 and can give this a look for you when the game isn't having maintenance.